ygot/render.go

```diff
@@ -884,6 +888,66 @@ func ConstructInternalJSON(s GoStruct) (map[string]interface{}, error) {
 	})
 }
 
+// Marshal7951Arg is an interface implemented by arguments to
+// the Marshal7951 function.
+type Marshal7951Arg interface {
+	// IsMarshal7951Arg is a market method.
+	IsMarshal7951Arg()
+}
+
+// JSONIndent is a string that specifies the indentation that should be used
+// for JSON input.
+type JSONIndent string
+
+// IsMarshal7951Arg marks JSONIndent as a valid Marshal7951 argument.
+func (JSONIndent) IsMarshal7951Arg() {}
+
+// Marshal7951 renders the supplied interface to RFC7951-compatible JSON. The argument
+// supplied must be a valid type within a generated ygot GoStruct - but can be a member
+// field of a generated struct rather than the entire struct - allowing specific fields
+// to be rendered. The supplied arguments control the JSON marshalling behaviour - both
+// base JSON Marshal (e.g., indentation), as well as RFC7951 specific options such as
+// YANG module names being appended.
+// The rendered JSON is returned as a byte slice - in common with json.Marshal.
+func Marshal7951(d interface{}, args ...Marshal7951Arg) ([]byte, error) {
+	var (
+		rfccfg *RFC7951JSONConfig
```

Comment:
big nit...can you make this var rfcCfg just to help readability a little.

Reply:
Of course -- done!